diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_acl.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_acl.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_acl.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_acl.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_acl.h>
 #include <vlan.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>


diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_cpuTag.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_cpuTag.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_cpuTag.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_cpuTag.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_cpuTag.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>

 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_dos.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_dos.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_dos.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_dos.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_dos.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_dot1x.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_dot1x.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_dot1x.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_dot1x.c
@@ -22,7 +22,7 @@
 #include <dal/rtl8373/dal_rtl8373_dot1x.h>
 #include <dal/rtl8373/dal_rtl8373_vlan.h>
 #include <dal/rtl8373/dal_rtl8373_lut.h>
-#include <string.h>
+#include <linux/string.h>
 #include <rtl8373_asicdrv.h>
 //#include <rtl8373_asicdrv_rma.h>
 //#include <rtl8373_asicdrv_lut.h>

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_drv.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_drv.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_drv.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_drv.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal_rtl8373_drv.h>
-#include <string.h>
+#include <linux/string.h>
 #include <rtl8373_asicdrv.h>
 #include <phy_rtl8224.h>
 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_eee.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_eee.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_eee.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_eee.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_eee.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>

 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_fc.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_fc.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_fc.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_fc.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_fc.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_gpio.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_gpio.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_gpio.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_gpio.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_gpio.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>



diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_i2c.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_i2c.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_i2c.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_i2c.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_i2c.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>

 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_igmp.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_igmp.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_igmp.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_igmp.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_igmp.h>
-#include <string.h>
+#include <linux/string.h>

 #include <rtl8373_asicdrv.h>
 #include <dal_rtl8373_lut.h>

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_interrupt.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_interrupt.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_interrupt.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_interrupt.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_interrupt.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>




diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_isolation.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_isolation.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_isolation.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_isolation.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_isolation.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>


diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_led.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_led.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_led.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_led.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal_rtl8373_led.h>
-#include <string.h>
+#include <linux/string.h>
 #include <rtl8373_reg_definition.h>
 #include <rtl8373_asicdrv.h>


diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_lut.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_lut.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_lut.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_lut.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_lut.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>
 #include "l2.h"

 static void _rtl8373_fdbStUser2Smi( rtl8373_luttb *pLutSt, rtk_uint32 *pFdbSmi)

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_macsec.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_macsec.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_macsec.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_macsec.c
@@ -23,7 +23,7 @@
 #include <dal/rtl8373/dal_rtl8373_macsec_ip_reg.h>
 #include <dal/rtl8373/dal_rtl8373_drv.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>

 /* Function Name:
  *      dal_rtl8373_macsec_enable_set

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_phy.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_phy.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_phy.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_phy.c
@@ -3,7 +3,7 @@

 #include <dal_rtl8373_drv.h>
 //#include <stdlib.h>
-#include <string.h>
+#include <linux/string.h>
 #include <rtk_error.h>

 //#include "port.h"

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_port.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_port.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_port.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_port.c
@@ -23,7 +23,7 @@
 #include <dal_rtl8373_switch.h>
 #include <dal_rtl8373_drv.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>

 rtk_uint32 PORT3_PHYAD=0;
 rtk_uint32 PORT8_PHYAD=0;

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_ptp.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_ptp.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_ptp.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_ptp.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_ptp.h>
-#include <string.h>
+#include <linux/string.h>
 #include <miim.h>
 #include <rtl8373_asicdrv.h>
 //#include <rtl8373_asicdrv_eav.h>

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_qos.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_qos.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_qos.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_qos.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_qos.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>


diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_rate.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_rate.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_rate.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_rate.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_rate.h>
 #include <vlan.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>


diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_rma.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_rma.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_rma.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_rma.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_rma.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_sharemeter.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_sharemeter.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_sharemeter.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_sharemeter.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_sharemeter.h>
-#include <string.h>
+#include <linux/string.h>
 #include <sharemeter.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_storm.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_storm.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_storm.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_storm.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_storm.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/rtl8373/rtl8373_asicdrv.h>
 #include <dal/rtl8373/dal_rtl8373_rma.h>

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_stp.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.h b/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.h
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.h
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_stp.h
@@ -4,7 +4,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>

 enum RTL8373_MSTP_STATE
 {

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.c
@@ -23,7 +23,7 @@
 #include <dal/rtl8373/dal_rtl8373_vlan.h>
 #include <dal/rtl8373/dal_rtl8373_svlan.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.h b/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.h
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.h
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_svlan.h
@@ -20,7 +20,7 @@

 #include <svlan.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 /* Function Name:

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_switch.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_switch.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_switch.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_switch.c
@@ -23,7 +23,7 @@
 #include <dal/rtl8373/rtl8373_asicdrv.h>
 #include <dal_rtl8373_drv.h>
 #include <dal_rtl8373_port.h>
-#include <string.h>
+#include <linux/string.h>

 /*
  * Symbol Definition

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_trunk.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_trunk.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_trunk.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_trunk.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_trunk.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>




diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_vlan.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_vlan.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_vlan.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_vlan.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal/rtl8373/dal_rtl8373_vlan.h>
 #include <dal/rtl8373/rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>


 #if defined(CONFIG_RTL8373_ASICDRV_TEST)

diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.c b/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.c
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.c
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <dal_rtl8373_wol.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>




diff --git a/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.h b/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.h
--- a/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.h
+++ b/src/rtk-api/dal/rtl8373/dal_rtl8373_wol.h
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>




diff --git a/src/rtk-api/acl.c b/src/rtk-api/acl.c
--- a/src/rtk-api/acl.c
+++ b/src/rtk-api/acl.c
@@ -22,7 +22,7 @@
 #include <acl.h>
 #include <vlan.h>
 #include <svlan.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/cpuTag.c b/src/rtk-api/cpuTag.c
--- a/src/rtk-api/cpuTag.c
+++ b/src/rtk-api/cpuTag.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <cpuTag.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/dos.c b/src/rtk-api/dos.c
--- a/src/rtk-api/dos.c
+++ b/src/rtk-api/dos.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <trunk.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/dot1x.c b/src/rtk-api/dot1x.c
--- a/src/rtk-api/dot1x.c
+++ b/src/rtk-api/dot1x.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <dot1x.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/eee.c b/src/rtk-api/eee.c
--- a/src/rtk-api/eee.c
+++ b/src/rtk-api/eee.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <eee.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/gpio.c b/src/rtk-api/gpio.c
--- a/src/rtk-api/gpio.c
+++ b/src/rtk-api/gpio.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <gpio.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>



diff --git a/src/rtk-api/i2c.c b/src/rtk-api/i2c.c
--- a/src/rtk-api/i2c.c
+++ b/src/rtk-api/i2c.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <i2c.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>

 /* Function Name:

diff --git a/src/rtk-api/identify.c b/src/rtk-api/identify.c
--- a/src/rtk-api/identify.c
+++ b/src/rtk-api/identify.c
@@ -27,7 +27,7 @@
 #if defined(CONFIG_SDK_RTL8224)
 #include "phy_rtl8224.h"
 #endif
-#include <string.h>
+#include <linux/string.h>

 #if defined(CONFIG_SDK_RTL8224)
 static rt_phyInfo_t rtl8224_hal_Ctrl =

diff --git a/src/rtk-api/igmp.c b/src/rtk-api/igmp.c
--- a/src/rtk-api/igmp.c
+++ b/src/rtk-api/igmp.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <igmp.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>
 #include <dal/rtl8373/dal_rtl8373_igmp.h>

diff --git a/src/rtk-api/interrupt.c b/src/rtk-api/interrupt.c
--- a/src/rtk-api/interrupt.c
+++ b/src/rtk-api/interrupt.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <interrupt.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/isolation.c b/src/rtk-api/isolation.c
--- a/src/rtk-api/isolation.c
+++ b/src/rtk-api/isolation.c
@@ -20,7 +20,7 @@
  #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <isolation.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/l2.c b/src/rtk-api/l2.c
--- a/src/rtk-api/l2.c
+++ b/src/rtk-api/l2.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <l2.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/led.c b/src/rtk-api/led.c
--- a/src/rtk-api/led.c
+++ b/src/rtk-api/led.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <led.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>



diff --git a/src/rtk-api/macsec.c b/src/rtk-api/macsec.c
--- a/src/rtk-api/macsec.c
+++ b/src/rtk-api/macsec.c
@@ -18,7 +18,7 @@
  #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <macsec.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/mib.c b/src/rtk-api/mib.c
--- a/src/rtk-api/mib.c
+++ b/src/rtk-api/mib.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <mib.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/miim.c b/src/rtk-api/miim.c
--- a/src/rtk-api/miim.c
+++ b/src/rtk-api/miim.c
@@ -23,7 +23,7 @@
 #include "ptp.h"
 #include "phydef.h"

-#include <string.h>
+#include <linux/string.h>
 /* Function Name:
  *      phy_ptpRefTime_set
  * Description:

diff --git a/src/rtk-api/mirror.c b/src/rtk-api/mirror.c
--- a/src/rtk-api/mirror.c
+++ b/src/rtk-api/mirror.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <mirror.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>



diff --git a/src/rtk-api/nic.c b/src/rtk-api/nic.c
--- a/src/rtk-api/nic.c
+++ b/src/rtk-api/nic.c
@@ -22,7 +22,7 @@
 #include <rtk_error.h>
 #include <nic.h>

-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/phy.c b/src/rtk-api/phy.c
--- a/src/rtk-api/phy.c
+++ b/src/rtk-api/phy.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <phy.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>
 #include <rtk_types.h>


diff --git a/src/rtk-api/phy_rtl8224.c b/src/rtk-api/phy_rtl8224.c
--- a/src/rtk-api/phy_rtl8224.c
+++ b/src/rtk-api/phy_rtl8224.c
@@ -27,7 +27,7 @@
 #include <phy_rtl8224.h>
 #include <dal_rtl8373_ptp.h>
 #include <rtl8373_reg_definition.h>
-#include <string.h>
+#include <linux/string.h>
 /*
  * Data Declaration
  */

diff --git a/src/rtk-api/port.c b/src/rtk-api/port.c
--- a/src/rtk-api/port.c
+++ b/src/rtk-api/port.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <port.h>
 #include <rtl8373_asicdrv.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>



diff --git a/src/rtk-api/ptp.c b/src/rtk-api/ptp.c
--- a/src/rtk-api/ptp.c
+++ b/src/rtk-api/ptp.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <ptp.h>
-#include <string.h>
+#include <linux/string.h>
 #include <dal/dal_mgmt.h>

 /* Function Name:

diff --git a/src/rtk-api/qos.c b/src/rtk-api/qos.c
--- a/src/rtk-api/qos.c
+++ b/src/rtk-api/qos.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <qos.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/rate.c b/src/rtk-api/rate.c
--- a/src/rtk-api/rate.c
+++ b/src/rtk-api/rate.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <rate.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/rma.c b/src/rtk-api/rma.c
--- a/src/rtk-api/rma.c
+++ b/src/rtk-api/rma.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <rma.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/rtk_switch.c b/src/rtk-api/rtk_switch.c
--- a/src/rtk-api/rtk_switch.c
+++ b/src/rtk-api/rtk_switch.c
@@ -18,7 +18,7 @@
  */
 #include <rtk_switch.h>
 #include <rtk_error.h>
-#include <string.h>
+#include <linux/string.h>
 #include <identify.h>
 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/sharemeter.c b/src/rtk-api/sharemeter.c
--- a/src/rtk-api/sharemeter.c
+++ b/src/rtk-api/sharemeter.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <sharemeter.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/storm.c b/src/rtk-api/storm.c
--- a/src/rtk-api/storm.c
+++ b/src/rtk-api/storm.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <storm.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/svlan.c b/src/rtk-api/svlan.c
--- a/src/rtk-api/svlan.c
+++ b/src/rtk-api/svlan.c
@@ -21,7 +21,7 @@
 #include <rtk_error.h>
 #include <svlan.h>
 #include <vlan.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/trunk.c b/src/rtk-api/trunk.c
--- a/src/rtk-api/trunk.c
+++ b/src/rtk-api/trunk.c
@@ -20,7 +20,7 @@
 #include <rtk_switch.h>
 #include <rtk_error.h>
 #include <trunk.h>
-#include <string.h>
+#include <linux/string.h>

 #include <dal/dal_mgmt.h>


diff --git a/src/rtk-api/vlan.c b/src/rtk-api/vlan.c
--- a/src/rtk-api/vlan.c
+++ b/src/rtk-api/vlan.c
@@ -23,7 +23,7 @@
 #include <rtk_error.h>
 #include <vlan.h>
 //#include <rate.h>
-#include <string.h>
+#include <linux/string.h>
 
 #include <dal/dal_mgmt.h>
 
